includes/wp-cli/class-one-time-fixers.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Automattic\WP\Cron_Control\CLI;
+
+class One_Time_Fixers extends \WPCOM_VIP_CLI_Command {
+
+	/**
+	 * Remove corrupt Cron Control data resulting from initial plugin deployment
+	 *
+	 * eg.: `wp --allow-root cron-control remove-all-plugin-data --batch-size=15 --dry-run=true`
+	 *
+	 * @subcommand remove-all-plugin-data
+	 */
+	public function purge( $args, $assoc_args ) {
+		global $wpdb;
+
+		// Are we actually destroying any data?
+		$dry_run = true;
+
+		if ( isset( $assoc_args['dry-run'] ) && 'false' === $assoc_args['dry-run'] ) {
+			$dry_run = false;
+		}
+
+		// Provide some idea of what's going on
+		\WP_CLI::line( __( 'CRON CONTROL', 'automattic-cron-control' ) . "\n" );
+
+		$count = $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(ID) FROM {$wpdb->posts} WHERE post_type = %s;", 'a8c_cron_ctrl_event' ) );
+
+		if ( is_numeric( $count ) ) {
+			$count = (int) $count;
+			\WP_CLI::line( sprintf( __( 'Found %s total items', 'automattic-cron-control' ), number_format_i18n( $count ) ) . "\n\n" );
+		} else {
+			\WP_CLI::line( __( 'Something went wrong...aborting!', 'automattic-cron-control' ) );
+			return;
+		}
+
+		// Should we really destroy all this data?
+		if ( ! $dry_run ) {
+			\WP_CLI::line( __( 'This process will remove all CPT data for the Cron Control plugin', 'automattic-cron-control' ) );
+			\WP_CLI::confirm( __( 'Proceed?', 'automattic-cron-control' ) );
+			\WP_CLI::line( "\n" . __( 'Starting...', 'automattic-cron-control' ) . "\n" );
+		}
+
+		// Determine how many batches this will take
+		if ( isset( $assoc_args['batch-size'] ) ) {
+			$page_size = max( 1, min( absint( $assoc_args['batch-size'] ), 500 ) );
+		} else {
+			$page_size = 250;
+		}
+		\WP_CLI::line( sprintf( __( 'Processing in batches of %s', 'automattic-cron-control' ), number_format_i18n( $page_size ) ) . "\n\n" );
+
+		$pages     = 1;
+		$page      = 1;
+
+		if ( $count > $page_size ) {
+			$pages = ceil( $count / $page_size );
+		}
+
+		// Let's get on with it
+		do {
+			\WP_CLI::line( "\n\n" . sprintf( __( 'Processing page %s of %s', 'automattic-cron-control' ), number_format_i18n( $page ), number_format_i18n( $pages ) ) . "\n" );
+
+			$items = $wpdb->get_results( $wpdb->prepare( "SELECT ID, post_title FROM {$wpdb->posts} WHERE post_type = %s LIMIT %d", 'a8c_cron_ctrl_event', $page_size ) );
+
+			// Nothing more to do
+			if ( ! is_array( $items ) || empty( $items ) ) {
+				\WP_CLI::line( __( 'No more items found!', 'automattic-cron-control' ) );
+				break;
+			}
+
+			\WP_CLI::line( sprintf( __( 'Found %s items in this batch' ), number_format_i18n( count( $items ) ) ) );
+
+			if ( ! $dry_run ) {
+				foreach ( $items as $item ) {
+					\WP_CLI::line( "{$item->ID}, `{$item->post_title}`" );
```

Comment:
Heh, yes, but I was being lazy and hoped you wouldn't notice. 😉

Since the query doesn't deal with pagination, the output during a dry-run doesn't change, but I'll move this back up anyway. Some output is better than nothing.